api/src/main/java/javax/enterprise/inject/spi/BeanManager.java

```diff
@@ -406,14 +408,25 @@
      * @throws ContextNotActiveException if there is no active context object for the given scope
      * @throws IllegalArgumentException if there is more than one active context object for the given scope
      */
-    public Context getContext(Class<? extends Annotation> scopeType);
+    Context getContext(Class<? extends Annotation> scopeType);
+
+    /**
+     * Creates a new {@link ManageableContext} that represents the given built-in scope.  This ManageableContext is not
+     * yet active, it must be activated when ready to use.  Upon activation, you may use it to retrieve beans from
+     * the context.
+     *
+     * @param scopeType the  {@linkplain javax.enterprise.context scope}
+     * @return the {@linkplain ManageableContext context object}
+     * @throws IllegalArgumentException if the given scopeType is not a valid built in scope.
+     */
+    ManageableContext createManageableContext(Class<? extends Annotation> scopeType);
```

Comment:
@johnament 
> This method specifically creates a new ManageableContext, it's not bound to anything per se.

Ok, it's not bound to anything until activated - after activation it must be "bound" to the current thread. Otherwise, it would contradict the spec and a user would not be able to inject normal scoped beans without the reference to the context object... 

@antoinesd Yep, it's quite complicated topic. And an ideal topic for F2F BTW ;-)